PacketDotNet/IPv6Packet.cs

```diff
@@ -299,14 +322,27 @@ public IPv6Packet(ByteArraySegment bas)
             Log.DebugFormat("PayloadLength: {0}", PayloadLength);
             Header.Length = bas.Length - PayloadLength;
 
-            // parse the payload
-            PayloadPacketOrData = new Lazy<PacketOrByteArraySegment>(() =>
+            if (PayloadLength > 0)
             {
-                var payload = Header.EncapsulatedBytes(PayloadLength);
-                return ParseEncapsulatedBytes(payload,
-                                              NextHeader,
-                                              this);
-            }, LazyThreadSafetyMode.PublicationOnly);
+                // parse the payload
+                PayloadPacketOrData = new Lazy<PacketOrByteArraySegment>(() =>
+                {
+                    //Strip off the Extension headers
```

Comment:
Instead of stripping them off entirely, I think those headers should still be accessible (see previous comment), similar to TcpOptions.

Reply:
Bad comment as they aren't being scripted off, just moving the pointer down to the next position

Reply:
Well, they aren't stripped off as such, but the data is now made inaccessible. If you were to implement a similar structure as TcpOptions then you'd still be able to access the headers.